public/app/plugins/datasource/grafana-pyroscope-datasource/datasource.test.ts

```diff
@@ -46,16 +33,21 @@ export function mockFetchPyroscopeDatasourceSettings(
   });
 }
 
-describe('Pyroscope data source', () => {
-  let ds: PyroscopeDataSource;
-  beforeEach(() => {
-    mockFetchPyroscopeDatasourceSettings();
-    setPluginExtensionGetter(() => ({ extensions: [] })); // No extensions
-    ds = new PyroscopeDataSource(defaultSettings);
-  });
+function setupDatasource() {
```

Comment:
Had to refactor this because otherwise there was some side-effects of running all the tests that seeped in this one and made it fail. I assume some cleanup somewhere wasn't done and affected the `getTemplateSrv` mock or something like that. This is more explicit so should be less prone to that.

Reply:
I like `setupDatasource`, it's definitely clearer.

Nit: could we use it in this way:
```
let ds: PyroscopeDataSource;
  beforeEach(() => {
    ds = setupDatasource();
  });
``` 
and remove the `ds = setupDatasource();` from all the tests, or do you think it would still cause problems and/or be less clear?

Reply:
If there was some super elaborate setup than maybe but I think it's weird to setup a global variable, set it up just before a test so each test does not leak to other tests. Like you use `ds` variable and to know what it is you have to jump to it's declaration where it's empty and then to `beforeEach` function to know how it is initialized. Personally I feel it does not add anything just makes the code harder to read.

Reply:
OK then let's keep it as it is, mine wasn't a strong opinion.